ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartUploadPartHandler.java

```diff
@@ -80,13 +213,88 @@ void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
         // Set S3 ETag header
         String blobId = (String) restResponseChannel.getHeader(LOCATION);
         restResponseChannel.setHeader("ETag", blobId);
-        //TODO [S3]: remove x-ambry- headers
-        callback.onCompletion(null, null);
-      } catch (RestServiceException e) {
-        callback.onCompletion(null, e);
+        finalCallback.onCompletion(null, null);
+      }, restRequest.getUri(), logger, finalCallback);
+    }
+
+    /**
+     * Parse {@link BlobInfo} from the request arguments. This method will also ensure that the correct account and
+     * container objects are attached to the request.
+     * @return the {@link BlobInfo} parsed from the request arguments.
+     * @throws RestServiceException if there is an error while parsing the {@link BlobInfo} arguments.
+     */
+    private BlobInfo getBlobInfoFromRequest() throws RestServiceException {
+      // TODO [S3] may consolidate and move it to a central place.
+      long propsBuildStartTime = System.currentTimeMillis();
+      accountAndContainerInjector.injectAccountContainerForNamedBlob(restRequest, frontendMetrics.putBlobMetricsGroup);
+      if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
+        accountAndContainerInjector.injectDatasetForNamedBlob(restRequest);
+      }
+      BlobProperties blobProperties = RestUtils.buildBlobProperties(restRequest.getArgs());
+      Container container = RestUtils.getContainerFromArgs(restRequest.getArgs());
+      if (blobProperties.getTimeToLiveInSeconds() + TimeUnit.MILLISECONDS.toSeconds(
+          blobProperties.getCreationTimeInMs()) > Integer.MAX_VALUE) {
+        logger.debug("TTL set to very large value in PUT request with BlobProperties {}", blobProperties);
+        frontendMetrics.ttlTooLargeError.inc();
+      } else if (container.isTtlRequired() && (blobProperties.getTimeToLiveInSeconds() == Utils.Infinite_Time
```

Comment:
Since TTL for S3 multipart requests would be set by us, may be we don't need these `if/else if` checks?

Reply:
Removed all the Ttl related code. Thanks